pubnub/endpoints/message_actions/get_message_actions.py

```diff
@@ -1,35 +1,42 @@
 from pubnub import utils
 from pubnub.endpoints.endpoint import Endpoint
+from pubnub.models.consumer.common import PNStatus
 from pubnub.models.consumer.message_actions import PNGetMessageActionsResult, PNMessageAction
 from pubnub.enums import HttpMethod, PNOperationType
+from pubnub.structures import Envelope
+
+
+class PNGetMessageActionsResultEnvelope(Envelope):
+    result: PNGetMessageActionsResult
+    status: PNStatus
 
 
 class GetMessageActions(Endpoint):
     GET_MESSAGE_ACTIONS_PATH = '/v1/message-actions/%s/channel/%s'
     MAX_LIMIT = 100
 
-    def __init__(self, pubnub):
+    def __init__(self, pubnub, channel: str = None, start: str = None, end: str = None, limit: str = None):
```

Comment:
I assume it's not possible to change it, so let's leave it. But I noticed that perhaps all responses contain reference to a PubNub object. Is there any benefit from such hard coupling?

Reply:
it's better than having a global variable because you may have more than one instance running at the same time